sql/update.go

```diff
@@ -333,6 +333,9 @@ func (p *planner) Update(n *parser.Update, autoCommit bool) (planNode, *roachpb.
 				if _, ok := colIDSet[col.ID]; ok {
 					continue
 				}
+				if _, ok := primaryKeyCols[col.ID]; ok {
+					continue
+				}
```

Comment:
No. We're in an if-branch here which is only executed when one of the primary-key columns changes. If you're changing a non-primary key column we only touch the column key and index keys that are affected. For column keys that translates to a `Put`, but for index keys we have to perform a `Del` followed by a `CPut`.